### ISSUE TITLE
ch. "Files, Modules, and Programs" could mention private types?

### ISSUE THREAD

**Issue by ulugbekna** [realworldocaml/book#3391]
Chapter "Files, Modules, and Programs" doesn't mention private types as a somewhat middle ground between abstract and concrete types. I feel like private types could contribute to the discussion in this chapter, especially in the "Designing with modules" part. 

Sidenote: Thanks for an awesome book! :-)

**Comment by yminsky** [comment]
Yeah, that's a good idea. Private types, in their various forms, would be good to cover.  It's a bit of a tricky topic, but I'll throw it on the list.
